ambry-network/src/main/java/com/github/ambry/network/http2/Http2BlockingChannel.java

```diff
@@ -108,14 +106,25 @@ public ChannelOutput sendAndReceive(Send request) throws IOException {
 
     Promise<ByteBuf> responsePromise = streamChannel.eventLoop().newPromise();
     streamChannel.attr(RESPONSE_PROMISE).set(responsePromise);
-    streamChannel.writeAndFlush(request)
+    boolean success = streamChannel.writeAndFlush(request)
         .awaitUninterruptibly(http2ClientConfig.http2BlockingChannelSendTimeoutMs, TimeUnit.MILLISECONDS);
+    if (!success) {
+      streamChannel.parent()
```

Comment:
In Http2BlockingChannelResponseHandler we have logic to ensure that release is only called once (`getAndSet(null)` call). Do we need that same logic here? Could there be a race condition between the handler getting a response/exception and this timeout?